lib/comma/data_extractor.rb

```diff
@@ -29,12 +29,25 @@ def yield_block_with_value(value, &block)
       end
 
       def extract_value(method)
-        extraction_object.send(method)
+        output = extraction_object.send(method)
+        time_zone_adjusted_value(output)
       end
 
       def extraction_object
         @instance
       end
+
+      def should_adjust_time_zone?(object)
+        object.respond_to?(:in_time_zone) && extraction_object.respond_to?(:site)
```

Comment:
To make this a little more generic, could this check ":current_time_zone" directly, instead of trying to call "site".  And then any object that wants to have "smart" adjustments should just delegate that method to "site" (or somewhere else if it's more appropriate)